mlir/lib/Conversion/SCFToEmitC/SCFToEmitC.cpp

```diff
@@ -182,18 +202,22 @@ LogicalResult IfLowering::matchAndRewrite(IfOp ifOp,
   return success();
 }
 
-void mlir::populateSCFToEmitCConversionPatterns(RewritePatternSet &patterns) {
-  patterns.add<ForLowering>(patterns.getContext());
-  patterns.add<IfLowering>(patterns.getContext());
+void mlir::populateSCFToEmitCConversionPatterns(RewritePatternSet &patterns,
+                                                TypeConverter &typeConverter) {
+  patterns.add<ForLowering>(typeConverter, patterns.getContext());
+  patterns.add<IfLowering>(typeConverter, patterns.getContext());
 }
 
 void SCFToEmitCPass::runOnOperation() {
   RewritePatternSet patterns(&getContext());
-  populateSCFToEmitCConversionPatterns(patterns);
+  TypeConverter typeConverter;
+  populateEmitCSizeTypeConversions(typeConverter);
+  populateEmitCDefaultTypeConversions(typeConverter);
```

Comment:
Why are the size type conversions and the default type conversions separate?

Reply:
We don't make an assumption on where the type conversions are going to be used, so we don't want to provide an identity type conversion for types that may need to be converted otherwise. Passes usually add these default converters excluding the types they know are converted (look at [MemRefToEmitC](https://github.com/Xilinx/llvm-project/pull/186/files#diff-a009fa890763babc60743e4cb57af0c30897b0783b110d4d6636bb3820b8c249R36) for instance).

SCFToEmitC, ArithToEmitC and FuncToEmitC all convert the same types, so they all have the same default conversions. This function lets us have it written only once (and not go through them again when we'll do booleans and signless integers).